src/components/Books.jsx

```diff
@@ -1,32 +1,25 @@
-import PropTypes from 'prop-types';
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { fetchBooks } from '../redux/books/booksSlice';
 import AddNewBook from './AddNewBook';
 import Book from './Book';
 
-const Books = ({ books, addBook, deleteBook }) => (
+const Books = () => {
+  const dispatch = useDispatch();
+  const books = useSelector((state) => state.books.books);
 
-  <div className="books-div">
-    {books.length > 0 ? books.map((book) => (
-      <Book
-        book={book}
-        key={book.item_id}
-        deleteBook={deleteBook}
-      />
-    )) : (
-      <p className="no-books">No books yet. Please add a book.</p>
-    )}
-    <hr />
-    <AddNewBook addBook={addBook} />
-  </div>
-);
-Books.propTypes = {
-  books: PropTypes.arrayOf(PropTypes.shape({
-    item_id: PropTypes.string.isRequired,
-    category: PropTypes.string.isRequired,
-    title: PropTypes.string.isRequired,
-    author: PropTypes.string.isRequired,
-  })).isRequired,
-  addBook: PropTypes.func.isRequired,
-  deleteBook: PropTypes.func.isRequired,
+  useEffect(() => {
+    dispatch(fetchBooks());
+  });
```

Comment:
[OPTIONAL] This is an optional requirement but recommended so kindly don't skip this requirement please add a dependency  array after the curly braces of `useEffect` because if you don't pass the dependency array  then it's called again and again which is not suitable for performance and if you pass dependency array then it's call only one time when the page is loaded so kindly add a dependency  array after the curly braces of `useEffect` :slightly_smiling_face: 

```suggestion
  useEffect(() => {
    dispatch(fetchBooks());
  },[]);
```